tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -323,23 +326,22 @@ void test_all_channels_instance(const char *str)
 	const int n = 10;
 	int nchan = 0;
 	bool limit_reached = false;
-	struct counter_alarm_cfg alarm_cfgs[n];
+	struct counter_alarm_cfg alarm_cfgs;
```

Comment:
The changes in this function do not seem to be related to a failing test. This is more refactoring, so I think it should be done in a separate commit (lines [209-210](https://github.com/zephyrproject-rtos/zephyr/pull/13305/files#diff-cbfda9aaae5d9069931b177bfff91d59R209) could be also removed in it, since they are a duplicate that probably popped up during some merging).